lib/auth/person.ex

```diff
@@ -0,0 +1,252 @@
+defmodule Auth.Person do
+  use Ecto.Schema
+  import Ecto.Changeset
+  alias Auth.Repo
+  # https://stackoverflow.com/a/47501059/1148249
+  alias __MODULE__
+
+  schema "people" do
+    field :auth_provider, :string
+    field :email, Fields.EmailEncrypted
+    field :email_hash, Fields.EmailHash
+    field :familyName, Fields.Encrypted
+    field :givenName, Fields.Encrypted
+    field :locale, :string
+    field :password, :string, virtual: true
+    field :password_hash, Fields.Password
+    field :picture, :binary
+    field :username, :binary
+    field :username_hash, Fields.Hash
+    field :status, :id
+    field :tag, :id
+    field :key_id, :integer
+
+    has_many :statuses, Auth.Status
+    # has_many :sessions, Auth.Session, on_delete: :delete_all
+    timestamps()
+  end
+
+  @doc """
+  Default attributes validation for Person
+  """
+  def changeset(person, attrs) do
+    person
+    |> cast(attrs, [
+      :id,
+      :username,
+      :email,
+      :givenName,
+      :familyName,
+      :password,
+      :password_hash,
+      :key_id,
+      :locale,
+      :picture,
+      :username,
+      :auth_provider,
+      :status
+    ])
+    |> validate_required([:email])
+    |> put_email_hash()
+    |> put_pass_hash()
+  end
+
+  def create_person(person) do
+    # IO.inspect(person, label: "create_person:51")
+    person =
+      %Person{}
+      |> changeset(person)
+      |> put_email_status_verified()
+      # |> IO.inspect(label: "after put_email_status_verified")
+
+    case get_person_by_email(person.changes.email) do
+      nil ->
+        Repo.insert!(person)
+
+      person ->
+        person
+    end
+  end
+
+  def login_register_changeset(attrs) do
+    %Person{}
+    |> cast(attrs, [:email])
+  end
+
+  def password_new_changeset(attrs) do
+    %Person{}
+    |> cast(attrs, [:email, :password])
+  end
+
+  @doc """
+  `transform_github_profile_data_to_person/1` transforms the profile data
+  received from invoking `ElixirAuthGithub.github_auth/1`
+  into a `person` record that can be inserted into the people table.
+
+  ## Example
+
+    iex> transform_profile_data_to_person(%{
+      avatar_url: "https://avatars3.githubusercontent.com/u/194400?v=4",
+      email: "alex@gmail.com",
+      followers: 2846,
+      login: "alex",
+      name: "Alex McAwesome",
+      type: "User",
+      url: "https://api.github.com/users/alex"
+    })
+    %{
+      "email" => "nelson@gmail.com",
+      "picture" => "https://avatars3.githubusercontent.com/u/194400?v=4",
+      "status" => 1,
+      "givenName" => "Alex McAwesome"
+    }
+  """
+  def transform_github_profile_data_to_person(profile) do
+    profile
+    |> Map.put(:username, profile.login)
+    |> Map.put(:givenName, profile.name)
+    |> Map.put(:picture, profile.avatar_url)
+    |> Map.put(:auth_provider, "github")
```

Comment:
I think we might be able to use `Map.merge` here

Reply:
@SimonLab yes, definitely. 👍